### PR TITLE
chore: WIP! Adding printer

### PR DESCRIPTION
**Pull request by KFlash**
Implements a simple pretty printer for Kataw AST. 

The returned output is similar to Prettier except for the known Prettier bugs and comment issues.

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add Do you see what's coming next?

**Comment by KFlash** [comment]
@aladdin-add You has to take an decision here. Should the comments be on or off by default?

**Comment by KFlash** [comment]
@aladdin-add  Comments will be printed by default. I think it's okay?

This is how I print a "simple" CST node. The "comment magic" already fixed :)

```ts
function printForOfStatement(node: any, printer: Printer):  ForOfStatement {
  return chain([
    printKeyword(node.forKeyword, printer, node.start, node),
    printKeyword(node.awaitKeyword, printer, node.forKeyword.end, node),
    printer.space,
    printPunctuator('(', node.awaitKeyword ? node.awaitKeyword.end : node.forKeyword.end, node),
    printForBinding(node.initializer, printer, node),
    printer.space,
    printKeyword(node.ofKeyword, printer, node.initializer.end, node),
    printer.space,
    printExpressions(node.expression, printer, node),
    printPunctuator(')', node.initializer.end, node),
    adjustClause(node.statement, printer, printStatements(node.statement, printer, node))
  ]);
}
```

**Comment by KFlash** [comment]
@aladdin-add the 'container' is a little tricky, maybe finish that one in a separate PR? It's the "outer wrapper" for comments on statement level and detach comments. 

The CST node printing almost ready to be pushed

**Comment by KFlash** [comment]
@aladdin-add Looks good!

looks_good

**Comment by KFlash** [comment]
60% has not yet been ported yet from my old code base, but the printer is there now.

**Comment by KFlash** [comment]
@aladdin-add It still prints invalid syntax. That's a HUGE problem. See #68

**Comment by aladdin-add** [comment]
there is still a test failing: https://github.com/kataw/kataw/blob/printer/test/__snapshot__/parser/expressions/template/crlf.md#input

the reason behind it: the printed code contains `\r\n`, but the repo has configured git to `\n`:
https://github.com/kataw/kataw/blob/26b5d903f19a9f570445dc2939194ba2604e2f6b/.gitattributes#L2
so, before committing to the repo, git would convert `\r\n` to `\n` automaticly.

**Comment by KFlash** [comment]
How can we solve that test? Or just remove it?

**Comment by KFlash** [comment]
@aladdin-add I'm little confused. What should we do with literals? Should we by default use `rawText` or `text` ? Maybe optional?
Linebreaks and trailing commas? I think the best solution is to use linebreaks if the 'NodeFlags' contains that masks, and skip it for the minified output?

**Comment by KFlash** [comment]
@aladdin-add can you resolve the conflicts?

**Comment by KFlash** [comment]
@aladdin-add Can you refresh the tests and solve the conflict?

**Review by aladdin-add — APPROVED** [review]

**Comment by KFlash** [comment]
@aladdin-add What we do? I clean up the source and we merge? Then I finish the comments in another PR ?

**Comment by aladdin-add** [comment]
agreed, the PR is getting too big, and hard to follow. :)

**Comment by KFlash** [comment]
Okay. Within this day I should have cleaned up this PR. Do you like the printer?
